src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c

```diff
@@ -0,0 +1,542 @@
+/*
+ * This file is part of the Soletta (TM) Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <sol-util-internal.h>
+#include <errno.h>
+
+#include <sol-iio.h>
+#include <sol-log.h>
+#include <sol-mainloop.h>
+
+/* #undef MQTT if you don't plan to use it */
+#ifdef MQTT
+#include <sol-mqtt.h>
+#endif
+
+/* comment out following if you don't plan to use calibration and denoise */
+#define GYRO_CALIBRATE
+#define DENOISE_MEDIAN
+
+#ifdef GYRO_CALIBRATE
+#include <math.h>
+#define GYRO_MAX_ERR 0.05
+#define GYRO_DS_SIZE 100
+#endif
+
+#ifdef DENOISE_MEDIAN
+#define GYRO_DENOISE_MAX_SAMPLES 5
+#define GYRO_DENOISE_NUM_FIELDS 3
+#endif
+
+#define GYRO_DROP_SAMPLES 5 /* Drop first few gyro samples for noisy sensor */
+
+#ifdef GYRO_CALIBRATE
+struct gyro_cal_t {
+    bool calibrated;
+    double bias_x, bias_y, bias_z;
+    int count;
+    double min_x;
+    double min_y;
+    double min_z;
+    double max_x;
+    double max_y;
+    double max_z;
+};
+#endif
+
+#ifdef DENOISE_MEDIAN
+struct filter_median_t {
+    double *buff;
+    unsigned int idx;
+    unsigned int count;
+    unsigned int sample_size;
+};
+#endif
+
+struct iio_gyroscope_data {
+    struct sol_iio_channel *channel_x;
+    struct sol_iio_channel *channel_y;
+    struct sol_iio_channel *channel_z;
+#ifdef MQTT
+    struct sol_mqtt *mqtt;
+    char *mqtt_topic;
+#endif
+#ifdef GYRO_CALIBRATE
+    struct gyro_cal_t cal_data;
+#endif
+#ifdef DENOISE_MEDIAN
+    struct filter_median_t filter_data;
+#endif
+    unsigned int drop_samples_count;
+};
+
+#ifdef MQTT
+static bool
+try_reconnect(void *data)
+{
+    SOL_INF("Try reconnect...");
+    return sol_mqtt_reconnect((struct sol_mqtt *)data) != 0;
+}
+
+static void
+on_connect(void *data, struct sol_mqtt *mqtt)
+{
+    if (sol_mqtt_get_connection_status(mqtt) == SOL_MQTT_CONNECTED) {
+        SOL_INF("Connected...");
+    } else {
+        SOL_WRN("Unable to connect, retrying...");
+        sol_timeout_add(1000, try_reconnect, mqtt);
```

Comment:
I see https://github.com/solettaproject/soletta/blob/master/src/samples/mqtt/mqtt-publish.c#L73 have the global timer for the publish callback and free when program exit but https://github.com/solettaproject/soletta/blob/master/src/samples/mqtt/mqtt-publish.c#L66 not doing it. 

Do you think my sample need it because I am not using the timer to do publish_callback.